### ISSUE TITLE
Metered usage billing without tiers

### ISSUE THREAD

**Issue by jgillich** [tierrun/tier#223]
### What is the issue?

I'd like to implement metered usage billing without tiers, like:

![Screenshot from 2023-01-26 10-53-13](https://user-images.githubusercontent.com/347965/214806821-7c01661a-3d32-41d5-9752-6f8d368709fa.png)

`tier.price` allows floats, but `tier.base` and `base` do not. The reason is unclear to me, as this is fully supported by Stripe (as seen above).

### Steps to reproduce

pricing.json:

```
{
  "plans": {
    "plan:foo@1": {
      "title": "Foo",
      "currency": "eur",
      "features": {
        "feature:nano": {
          "title": "Foo",
          "base": 0.000001,
          "aggregrate": "sum"
        }
      }
    }
  }
}
```

```
tier: json: cannot unmarshal number 0.000001 into Go struct field Feature.plans.features.base of type int
```

### OS

Linux

### Tier version

0.7.1-dev

**Comment by jgillich** [comment]
After a bit more reading, it appears that usage-based billing is intended to be used with tiers only:

> When a feature has a "base" price, and no "tiers", then Tier will create the Price in Stripe as a flat rate, rather than usage-based.

https://www.tier.run/docs/mapping/

Is this correct? If so, this issue may be invalid. But Stripe doesn't require tiers here, so perhaps it could be supported...

**Comment by bmizerany** [comment]
@jgillich Stripe supports fractional unit amounts for metered pricing only. In pricing JSON, setting a base price without tiers signals the feature is to be billed as a licensed product in Stripe, and Stripe allows only an integer for a `unit_amount` for a licensed price.

See: https://stripe.com/docs/api/prices/object

> unit_amount: integer: The unit amount in cents to be charged, represented as a whole integer if possible. Only set if billing_scheme=per_unit.

Closing, but please reopen if I misunderstood your question.

**Comment by jgillich** [comment]
With the "Standard pricing" model in Stripe, I can tick the "usage is metered" to get metered billing without tiers, and it does allow fractional prices.

![Screenshot from 2023-02-23 01-46-44](https://user-images.githubusercontent.com/347965/220796885-854ed7e9-aa81-4dd2-b68b-ea2445e4d9d6.png)


I've not tried using the API to create prices, but I believe it's as simple as setting `unit_amount_decimal` instead of `unit_amount`.

**Comment by jgillich** [comment]
Any idea if this could be supported @bmizerany? Or is the intention to support usage pricing with tiers only? I could use tiers, but it complicates the UI and probably also the invoices. So I'd like to keep it simple

**Comment by bmizerany** [comment]
I do think it's possible. I'm looking into this now.

**Comment by jgillich** [comment]
Hah that was quick, thank you!

**Comment by bmizerany** [comment]
@jgillich I landed support for fractional units in main. Do you have Go 1.20 installed? If so, do you mind testing:

```
; go run tier.run/cmd/tier@b80919af8c26a47afbbdad5f88463193fd464438 push pricing.json
```

**Comment by jgillich** [comment]
Works great :+1: 

But I noticed that even with `aggregrate=sum`, it does not result in metered billing being enabled. Is that an option currently?

**Comment by bmizerany** [comment]
> Works great :+1: 
> 
> 
> 
> But I noticed that even with `aggregrate=sum`, it does not result in metered billing being enabled. Is that an option currently?

Features (for now) are metered if and only if the have at least one tier; otherwise they are considered licensed.

If you're seeing that features with tires are not metered in Stripe, Do you have a pricing.json you can share?

**Comment by bmizerany** [comment]
@jgillich Regarding metered billing without tiers: are you saying you want to bill for a variable quantity of a feature at the beginning of a billing period? If so, if there are units paid for, but unused, at the start of the next period, how would you like them handled? Prorated/refunded?

**Comment by jgillich** [comment]
Subscription items are created with a quantity of zero, and usage is reported as it occurs. So there can't be paid but unused units.

**Comment by bmizerany** [comment]
@jgillich, I see where my confusion was. I'm working on support for this now.

**Comment by bmizerany** [comment]
@jgillich Do you mind helping test? :) https://github.com/tierrun/tier/pull/267#issuecomment-1449168331

**Comment by jgillich** [comment]
LGTM, thanks!

**Comment by bmizerany** [comment]
@jgillich Great! Closing as is fixed in #267
